deephyper/search/nas/model/space/op/gnn.py

```diff
@@ -0,0 +1,507 @@
+import tensorflow as tf
+from numpy.random import seed
+from tensorflow.keras import activations, initializers, regularizers, constraints
+from tensorflow.keras.layers import LeakyReLU, Dropout, Dense
+import tensorflow.keras.backend as K
+
+seed(2020)
+from tensorflow import set_random_seed
+
+set_random_seed(2020)
+
+
+def tensor_dot(X):
+    assert type(X) is list
+    out = K.dot(X[0], X[1])
+    if len(X) > 2:
+        for i in range(2, len(X)):
+            out = K.dot(out, X[i])
+    return out
+
+
+class Apply1DMask(tf.keras.layers.Layer):
+    def __init__(self):
+        super(Apply1DMask, self).__init__()
+        pass
+
+    def __str__(self):
+        return f"Apply a 1D mask to the dense layers"
+
+    def call(self, inputs, **kwargs):
+        assert type(inputs) is list
+        return tf.multiply(inputs[0], inputs[1])
+
+
+class GlobalMaxPool(tf.keras.layers.Layer):
+    def __init__(self, axis=-2, **kwargs):
+        super(GlobalMaxPool, self).__init__()
+        self.axis = axis
+        self.kwargs = kwargs
+
+    def __str__(self):
+        return f"GraphMaxPool"

```

Comment:
useless `f` statement, also using `type(self).__name__` si better for generarlization.